### PR TITLE
ppOpen-AT: New package

### PR DESCRIPTION
**Pull request by tkameyama**

### CODE REVIEW

File: var/spack/repos/builtin/packages/ppopen-at/package.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    """ppOpen-AT is a part of the ppOpenHPC"""
+
+    homepage = "http://ppopenhpc.cc.u-tokyo.ac.jp/ppopenhpc/"
+    url = "file://{0}/ppohAT_1.0.0.tar.gz".format(os.getcwd())
+
+    version('1.0.0', sha256='2b5664839762c941e0b2dd7c15416e2dcfd5d909558cf7e4347a79ce535f3887')
+
+    def edit(self, spec, prefix):
+        makefile_in = FileFilter('Makefile.in')
+        makefile_in.filter('gcc', spack_cxx)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ppopen-at/package.py]: It's not supposed to be `spack_cc`?

File: var/spack/repos/builtin/packages/ppopen-at/package.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        makefile_in.filter('~/ppohAT_1.0.0', prefix)
+        makefile_in.filter('mkdir', 'mkdir -p')
+
+    def install(self, spec, prefix):
+        make('install')
+        mkdir(join_path(prefix, 'examples'))
+        copy_tree('examples', join_path(prefix, 'examples'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ppopen-at/package.py]:
```suggestion
        install_tree('examples', prefix.examples)
```

File: var/spack/repos/builtin/packages/ppopen-at/package.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        mkdir(join_path(prefix, 'doc'))
+        copy_tree('doc', join_path(prefix, 'doc'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ppopen-at/package.py]:
```suggestion
        install_tree('doc', prefix.doc)
```

File: var/spack/repos/builtin/packages/ppopen-at/package.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    def edit(self, spec, prefix):
+        makefile_in = FileFilter('Makefile.in')
+        makefile_in.filter('gcc', spack_cxx)
+        makefile_in.filter('~/ppohAT_1.0.0', prefix)
+        makefile_in.filter('mkdir', 'mkdir -p')
+
+    def install(self, spec, prefix):
+        make('install')
+        mkdir(join_path(prefix, 'examples'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ppopen-at/package.py]: I don't think this `mkdir` is actually required, is it? `install_tree` should be able to handle non-existing installation directories.

File: var/spack/repos/builtin/packages/ppopen-at/package.py (hunk 1 above)
Comment (tkameyama) [reply var/spack/repos/builtin/packages/ppopen-at/package.py]: This package is written by C++.
Source filenames are  *.h and *.cpp.
But CC variable and gcc are used in Makefile.

File: var/spack/repos/builtin/packages/ppopen-at/package.py (hunk 4 above)
Comment (tkameyama) [reply var/spack/repos/builtin/packages/ppopen-at/package.py]: Fixed.
Thenk you.

**Pull request merged**